templates/tower-project.yaml

```diff
@@ -0,0 +1,354 @@
+              Effect: Allow
+              Principal:
+                AWS: !Ref S3ReadOnlyAccessArns
+              Action:
+                - "s3:ListBucket*"
+                - "s3:GetBucketLocation"
+                - "s3:GetObject"
+                - "s3:GetObjectAcl"
+              Resource:
+                - !Sub "arn:aws:s3:::${TowerBucket}"
+                - !Sub "arn:aws:s3:::${TowerBucket}/*"
+            - !Ref AWS::NoValue
+
+  TowerProjectConfigurationSecret:
+    Type: AWS::SecretsManager::Secret
+    Properties:
+      Name: !Sub "${AWS::StackName}-TowerProjectConfiguration"
+      Description: !Sub "Configuration for ${AWS::StackName} Project in Tower"
+      SecretString: !Sub
+        - >-
+          {
+            "forge_access_key_id":      "${ForgeAccessKeyId}",
+            "forge_secret_access_key":  "${ForgeSecretAccessKey}",
+            "forge_role_arn":           "${ForgeRoleArn}",
+            "bucket_name":              "${BucketName}",
+            "read_write_arns":          "${ReadWriteArns}",
+            "read_only_arns":           "${ReadOnlyArns}",
+            "vpc_id":                   "${VpcId}",
+            "subnet_ids":               "${SubnetIds}"
+          }
+        - ForgeAccessKeyId: !Ref TowerForgeServiceUserAccessKey
+          ForgeSecretAccessKey: !GetAtt TowerForgeServiceUserAccessKey.SecretAccessKey
+          ForgeRoleArn: !GetAtt TowerForgeServiceRole.Arn
+          BucketName: !Ref TowerBucket
+          ReadWriteArns: !Join [ ",", !Ref S3ReadWriteAccessArns ]
+          ReadOnlyArns: !Join [ ",", !Ref S3ReadOnlyAccessArns ]
+          VpcId: !ImportValue
+            'Fn::Sub': '${AWS::Region}-nextflow-vpc-VPCId'
+          SubnetIds: !Join
+            - ","
+            - - !ImportValue
+                'Fn::Sub': '${AWS::Region}-nextflow-vpc-PrivateSubnet'
+              - !ImportValue
+                'Fn::Sub': '${AWS::Region}-nextflow-vpc-PrivateSubnet1'
+              - !ImportValue
+                'Fn::Sub': '${AWS::Region}-nextflow-vpc-PrivateSubnet2'
+              - !ImportValue
+                'Fn::Sub': '${AWS::Region}-nextflow-vpc-PrivateSubnet3'
+      KmsKeyId: !GetAtt EncryptionKeyStack.Outputs.Key
+
+  TowerProjectConfigurationSecretPolicy:
+    Type: AWS::SecretsManager::ResourcePolicy
+    Condition: HasAnyS3AccessArns
+    Properties:
+      BlockPublicPolicy: True
+      SecretId: !Ref TowerProjectConfigurationSecret
+      ResourcePolicy:
+        Version: "2012-10-17"
+        Statement:
+          - !If
+            - HasS3ReadWriteAccessArns
+            - Sid: GrantReadAccessToS3ReadWriteAccessArns
+              Effect: Allow
+              Action: "secretsmanager:GetSecretValue"
+              Principal:
+                AWS: !Ref S3ReadWriteAccessArns
+              Resource: !Ref TowerProjectConfigurationSecret
+            - !Ref AWS::NoValue
+          - !If
+            - HasS3ReadWriteAccessArns
```

Comment:
should this be `HasS3ReadOnlyAccessArns` instead?

Reply:
Good catch!